src/android/SecureStorage.java

```diff
@@ -19,24 +35,82 @@
 import org.json.JSONObject;
 import org.json.JSONArray;
 
+import javax.crypto.IllegalBlockSizeException;
+
 public class SecureStorage extends CordovaPlugin {
+    private final ExecutorService threadPool = Executors.newCachedThreadPool();
     private static final String TAG = "SecureStorage";
 
     private static final boolean SUPPORTS_NATIVE_AES = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
     private static final boolean SUPPORTED = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
 
     private static final String MSG_NOT_SUPPORTED = "API 19 (Android 4.4 KitKat) is required. This device is running API " + Build.VERSION.SDK_INT;
     private static final String MSG_DEVICE_NOT_SECURE = "Device is not secure";
+    public static final String MIGRATED_FOR_SECURITY = "_SS_MIGRATED_FOR_SECURITY";
 
     private final Hashtable<String, SharedPreferencesHandler> SERVICE_STORAGE = new Hashtable<String, SharedPreferencesHandler>();
 
     private IntentRequestQueue intentRequestQueue;
 
     @Override
     protected void pluginInitialize() {
+
         super.pluginInitialize();
 
         intentRequestQueue = new IntentRequestQueue(this);
+
+    }
+
+
+    private void securityMigration() throws JSONException {
+        Log.e(TAG, "Migration Started");
+        //transfer all existing items to new table
+        Hashtable<Integer, TransitionValue> transitionTable = new Hashtable<Integer, TransitionValue>();
+        Hashtable<String,Boolean> RSAMap= new Hashtable<String, Boolean>();
+        Enumeration<String> services = SERVICE_STORAGE.keys();
+        while(services.hasMoreElements()){
+            String service = services.nextElement();
+            //initializing rsakeymapper
+            RSAMap.put(service, false);
+            SharedPreferencesHandler handler = SERVICE_STORAGE.get(service);
+            Set<String> keys = handler.keys();
+
+            for(String key : keys){
+                String value = handler.fetch(key);
+                String decrypted = decryptHelper(value, service);
+
+                TransitionValue t = new TransitionValue(service, key, decrypted);
+                SecureRandom i = new SecureRandom();
```

Comment:
Shouldn't the SecureRandom instance be reused for all calls?

Reply:
I want different Map keys for each transition value otherwise i won't be able to access them all